### PR TITLE
Queriing and Data manipulation (Database and Rails Terminal Updates)

### PR DESCRIPTION
**Pull request by stino-x**
Changes Made:
CRUD Operations:

Implemented Create, Read, Update, and Delete (CRUD) operations for posts and comments.
Custom Queries:
INCLUDED AN SQL FOLDER FOR YOU TO SEE QUERIES FOR YOUR CONVIENIENCE!!!!!!!

Added custom queries for enhanced data retrieval and manipulation.
Examples include finding posts by title, updating comment content, and proper error-handling for post deletion.
Associations:

Established associations between posts and comments for streamlined data navigation.
GitHub Links in HTML Tags:

Enhanced image tag accessibility by incorporating GitHub links.
Additional Information:
Screenshots:

Screenshot 1
Screenshot 2
Screenshot 3
Screenshot 4
Screenshot 5
Screenshot 6
Screenshot 7
Screenshot 8
Screenshot 9
Screenshot 10
Screenshot 11

Feel free to review and let me know if you'd like any adjustments or additions!

### CODE REVIEW

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @stino-x, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- All requested models have been added :+1: 


## Required Changes ♻️

- [ ] Hi @stino-x, nice job for generating the User, Post, Comment and Like models as requested. Right now I think the relationship between these models are not correctly set up. The User model should have many posts, comments, and likes with a custom method to retrieve the 3 most recent posts from the db. Once more double the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md) for this project. To correctly set up association between entities, kindly take a look at this [ERD](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png)

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb
Comment (aamirkhan2478) [inline app/models/user.rb:1]: - [ ] As mentioned by the previous :point_up:  reviewer, please add correct relationships but for now you add only add one relationship so kindly add the relationship as well :slightly_smiling_face: 

```rb
  has_many :likes
  has_many :comments
```

- [ ] Please add the above :point_up:  relationship as well :+1:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @stino-x, 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FrederickMih** [comment]
Hi @stino-x, please submit from your dashboard. I won't be able to approved or leave changes :smile: 

Screenshot from 2023-11-10 21-03-24

- But I think you have set up associations between models as requested and added all the custom methods :+1:

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @stino-x 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Set up associations between models.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**